x/subscription/keeper/subscription.go

```diff
@@ -154,7 +107,7 @@ func (k Keeper) CreateSubscription(
 		return utils.LavaFormatWarning("create subscription failed", err)
 	}
 
-	if !found || sub.AutoRenewal {
+	if !found || sub.AutoRenewalNextPlan != types.AUTO_RENEWAL_PLAN_NONE {
```

Comment:
sub.AutoRenewalNextPlan != types.AUTO_RENEWAL_PLAN_NONE i think is not needed anymore since we dont get here from the begin block

Reply:
True. Fixed in 1269175585ae3fbc37add86b0d59b43294e9abd4